### ISSUE TITLE
Exceptions from MessageFormat aren't caught

### ISSUE THREAD

**Issue by unok-kun** [checkstyle/checkstyle#581]
An invalid pattern in a message element of the config file (which is after all user input) causes the JRE's MessageFormat to throw an IllegalArgumentException when encountered, which is never caught in Checkstyle.

Sample config and code:
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
 
<module name="Checker">
  <property name="charset" value="UTF-8"/>
  <property name="severity" value="warning"/>
  <module name="TreeWalker">
    <module name="WhitespaceAround">
      <message key="ws.notFollowed"
          value="WhitespaceAround: ''{0}'' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)"/>
    </module>
  </module>
</module>
```
(Note the use of `{}` instead of the escaped form `'{}'`. MessageFormat's counterintuitive syntax doesn't exactly help.)
```
class Test {
  public static void main(String[] args) {
    System.out.println("a"+1);
  }
}
```

Actual result:
```
Exception in thread "main" java.lang.IllegalArgumentException: can't parse argument number: 
	at java.text.MessageFormat.makeFormat(MessageFormat.java:1429)
	at java.text.MessageFormat.applyPattern(MessageFormat.java:479)
	at java.text.MessageFormat.<init>(MessageFormat.java:362)
	at java.text.MessageFormat.format(MessageFormat.java:840)
	at com.puppycrawl.tools.checkstyle.api.LocalizedMessage.getCustomMessage(LocalizedMessage.java:296)
	at com.puppycrawl.tools.checkstyle.api.LocalizedMessage.getMessage(LocalizedMessage.java:262)
	at com.puppycrawl.tools.checkstyle.api.AuditEvent.getMessage(AuditEvent.java:109)
	at com.puppycrawl.tools.checkstyle.DefaultLogger.addError(DefaultLogger.java:101)
	at com.puppycrawl.tools.checkstyle.Checker.fireErrors(Checker.java:517)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:280)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:129)
Caused by: java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:592)
	at java.lang.Integer.parseInt(Integer.java:615)
	at java.text.MessageFormat.makeFormat(MessageFormat.java:1427)
	... 10 more
```

Expected result: The normal output should appear normally (as much as possible), since chances are this isn't the end user's fault. I suggest printing the raw message in place, since that's better than showing nothing or an error message; and then warning of the error in the config file elsewhere.
```
Starting audit...
ERROR: Invalid message syntax in configuration file, line 11: can't parse argument number
/path/to/Test.java:3:27: warning: '+' is not preceded with whitespace.
/path/to/Test.java:3:28: warning: WhitespaceAround: ''{0}'' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
Audit done.
```

**Comment by romani** [comment]
Exception is not user fault , but it does not mean that we should hide details of failure.
Exception is printed as full as possible to let user easily send it to developers. That will ease developers work on reproducing the case, as users usually send only errors, without any input file or config.
As exception happen there is no guaranty that Check works fine - so no violation messages from Check it is expected.
Few commits ago we started print name of file where exception is happen.
